pkg/controllers/dashboard/helm/repo.go

```diff
@@ -196,27 +202,36 @@ func (r *repoHandler) download(repoSpec *catalog.RepoSpec, status catalog.RepoSt
 	}
 
 	downloadTime := metav1.Now()
-	if repoSpec.GitRepo != "" && status.IndexConfigMapName == "" {
-		commit, err = git.Head(secret, metadata.Namespace, metadata.Name, repoSpec.GitRepo, repoSpec.GitBranch, repoSpec.InsecureSkipTLSverify)
-		if err != nil {
-			return status, err
+	if repoSpec.GitRepo != "" {
+		// the helm repo is hosted in a git repo
+		if status.IndexConfigMapName == "" {
+			// it is the first time to download this repo
+			commit, err = git.Head(secret, metadata.Namespace, metadata.Name, repoSpec.GitRepo, repoSpec.GitBranch, repoSpec.InsecureSkipTLSverify)
+			if err != nil {
+				return status, err
+			}
+		} else {
+			commit, err = git.Update(secret, metadata.Namespace, metadata.Name, repoSpec.GitRepo, repoSpec.GitBranch, repoSpec.InsecureSkipTLSverify)
+			if err != nil {
+				return status, err
+			}
+			// check if the IndexConfigMap exists
+			exist := true
+			_, err = r.configMaps.Get(status.IndexConfigMapNamespace, status.IndexConfigMapName, metav1.GetOptions{})
+			if err != nil && apierrors.IsNotFound(err) {
+				exist = false
+			}
+			if status.Commit == commit && exist {
+				status.DownloadTime = downloadTime
+				return status, nil
+			}
```

Comment:
I think it would be a good idea to consider the configmap as nonexistent if we get any error at all for the same of updating the `DownloadTime`.

```suggestion
			if err == nil && status.Commit == commit {
				status.DownloadTime = downloadTime
				return status, nil
			}
```

Also, if we get any other error besides `IsNotFound`, should we return the error and let the controller requeue?

Reply:
> I think it would be a good idea to consider the configmap as nonexistent if we get any error at all for the same of updating the DownloadTime.

the line for setting the downloadTime is at the bottom of the method for all other cases except the `status.Commit == commit && exist` case and the failure case. 

> Also, if we get any other error besides IsNotFound, should we return the error and let the controller requeue?

Good catch! I added a check for it.